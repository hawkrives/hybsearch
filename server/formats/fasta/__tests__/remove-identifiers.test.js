/* eslint-env jest */
'use strict'

const expect = require('expect')
const removeIdentifiers = require('../remove-identifiers')

test('removes the given identifiers from the data', () => {
	let input = `> Emydura_subglobosa__KC755190
ggaacaataaattatcacctcaaaagacac
> Emydura_victoriae__KC755189
ggaacaatcaattattaccccacaaagac
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa`

	let filteredSequences = removeIdentifiers(input, {
		nm: [
			[
				{ ident: 'Emydura_subglobosa__KC755190' },
				{ ident: 'Emydura_victoriae__KC755189' },
			],
		],
	})

	expect(filteredSequences).toMatchSnapshot()
})
