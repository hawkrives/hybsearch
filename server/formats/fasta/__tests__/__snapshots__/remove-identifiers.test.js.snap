// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removes the given identifiers from the data 1`] = `
"> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa"
`;
